bindings/go/src/examples/perftest/perftest.go

```diff
@@ -56,29 +56,48 @@ func getAddressOffsetForThread(t uint) unsafe.Pointer {
 }
 
 // Printing functions
+
+func printRule() {
+	dashes5 := strings.Repeat("-", 5)
+	dashes20 := strings.Repeat("-", 20)
+	fmt.Printf("%5s|%20s|%20s|\n", dashes5, dashes20, dashes20)
+}
+
 func printHeader() {
-	dashes := strings.Repeat("-", 20)
-	fmt.Printf("|%20s|%20s|%20s|%20s|\n", dashes, dashes, dashes, dashes)
-	fmt.Printf("|%20s|%20s|%20s|%20s|\n", "Thread", "Iteration", "Latency ", "Bandwidth (Gb/s)")
-	fmt.Printf("|%20s|%20s|%20s|%20s|\n", dashes, dashes, dashes, dashes)
+	printRule()
+	fmt.Printf("%5s|%20s|%20s|\n", "#", "Bandwidth (Mb/s)", "Messages")
+	printRule()
 }
 
-func printPerThreadStatistics(i uint, t uint) {
-	bw := float64(perfTestParams.messageSize) * float64(8e-9) / perfTest.completionTime[t].Seconds()
-	fmt.Printf("|%20s|%20s|%20s|%20f|\n", fmt.Sprintf("%v/%v", t+1, perfTestParams.numThreads),
-		fmt.Sprintf("%v/%v", i, perfTestParams.numIterations), perfTest.completionTime[t], bw)
+func printStatistics() {
+	i := 0
+	last := atomic.LoadInt32(&perfTest.numCompletedRequests)
+	ticker := time.NewTicker(time.Duration(perfTestParams.printInterval) * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			curr := atomic.LoadInt32(&perfTest.numCompletedRequests)
+			rate := (curr - last) / int32(perfTestParams.printInterval)
+			bw := float64(perfTestParams.messageSize) * float64(rate) * float64(1e-6)
+			fmt.Printf("%5d|%20f|%20d|\n", i, bw, rate)
```

Comment:
I suggest to print the current number of iterations. As it done in ucx_perftest. It helps to track progress of the test.
```suggestion
			fmt.Printf("%20d|%20f|%20d|\n", curr, bw, rate)
```

```
--------------------|--------------------|--------------------|
        # iterations|    Bandwidth (Mb/s)|            Messages|
--------------------|--------------------|--------------------|
               11647|        12317.622272|               11747|
               23612|        12546.211840|               11965|
               36386|        13394.509824|               12774|
               47363|        11510.218752|               10977|
               60787|        14076.084224|               13424|
               73025|        12832.473088|               12238|
               82980|        10438.574080|                9955|
               92376|         9852.420096|                9396|
--------------------|--------------------|--------------------|
```